### PR TITLE
Ci/centos

### PR DESCRIPTION
**Pull request by JasonMarechal25**
- Remove "CentOS" dockerfile
- Rename "CentOS-bare" dockerfile as "CentOS-system-deps"
- "CentOS-system-deps" dockefile provide just the system configuration to build on centOS. No third party dependency (boost, tbb, etc)
- Update ublish-Docker-Github-Action to use main branch instead of master branch
- CentOS build-system-deps action now use "CentOS-system-deps" container

### CODE REVIEW

File: .github/workflows/build_ubuntu.yml [hunk 1]
```diff
@@ -60,10 +55,7 @@ jobs:
       - name: Install mandatory system libraries
         run: |
           sudo apt-get update --fix-missing
-          sudo apt-get install libjsoncpp-dev libgtest-dev libboost-mpi-dev libboost-program-options-dev libtbb-dev
-          cd /usr/src/googletest/
-          sudo cmake .
-          sudo cmake --build .
+          sudo apt-get install -y libjsoncpp-dev libtbb-dev
```
Comment (a-zakir) [inline .github/workflows/build_ubuntu.yml]: did you move the install of `libgtest-dev libboost-mpi-dev libboost-program-options-dev` elsewhere?

Comment (JasonMarechal25) [reply .github/workflows/build_ubuntu.yml]: Since we build boost from source it is not necessary to install it. It prevents possible conflicts

**Review by tbittar — APPROVED** [review]

File: .github/workflows/build_centos7.yml [hunk 2]
```diff
@@ -117,8 +117,15 @@ jobs:
...
 
+      - name: Compile tbb
+        uses: ./.github/workflows/compile-tbb
+        with:
+          cmake: 'cmake3'
+
       - name: Install dependencies
         run: |
+          pip3 install --upgrade pip
+          pip3 install wheel #Does not work in requirements
```
Comment (a-zakir) [inline .github/workflows/build_centos7.yml:128]: is `wheel` a new dependency? why ?

File: .github/workflows/centos7-system-deps-build.yml [hunk 3]
```diff
@@ -86,19 +92,26 @@ jobs:
...
+        with:
+          cmake-version: '3.22.x'
+
       - name: Configure
         run: |
+          source /opt/rh/devtoolset-9/enable
           export LD_LIBRARY_PATH=/usr/lib64/openmpi/lib:$LD_LIBRARY_PATH
           export PATH=/usr/lib64/openmpi/bin:$PATH
-          cmake3 -B _build -S . \
+          cmake -B _build -S . \
```
Comment (a-zakir) [inline .github/workflows/centos7-system-deps-build.yml:105]: `cmake3` is used to build Boost and in .github/workflows/build_centos7.yml
could you explain what's the diff between them?

File: .github/workflows/centos7-system-deps-build.yml [hunk 4]
```diff
@@ -86,19 +92,26 @@ jobs:
...
           -DDEPS_INSTALL_DIR=rte-antares-deps-Release \
-          -DBUILD_TESTING=ON -DCMAKE_BUILD_TYPE=Release  -DCMAKE_INSTALL_PREFIX=_install -DBUILD_UI=ON -DALLOW_RUN_AS_ROOT=ON
+          -DBUILD_TESTING=ON -DCMAKE_BUILD_TYPE=Release  -DCMAKE_INSTALL_PREFIX=_install -DBUILD_UI=OFF -DALLOW_RUN_AS_ROOT=ON
```
Comment (a-zakir) [inline .github/workflows/centos7-system-deps-build.yml]: turning OFF UI will possible hide dependency problem e.g PyQt.

File: .github/workflows/centos7-system-deps-build.yml [hunk 5]
```diff
@@ -86,19 +92,26 @@ jobs:
...
 
       - name: Build
         run: |
+          source /opt/rh/devtoolset-9/enable
           export LD_LIBRARY_PATH=/usr/lib64/openmpi/lib:$LD_LIBRARY_PATH
           export PATH=/usr/lib64/openmpi/bin:$PATH
-          cmake3 --build _build --config Release -j2 --target install
+          cmake --build _build --config Release -j2 --target install
```
Comment (a-zakir) [inline .github/workflows/centos7-system-deps-build.yml:114]: same

File: .github/workflows/centos7-system-deps-build.yml (hunk 3 above)
Comment (JasonMarechal25) [reply .github/workflows/centos7-system-deps-build.yml:105]: None.
We're using the action jwlawson/actions-setup-cmake to install cmake[3], cmake is a simlink to cmake3

File: .github/workflows/centos7-system-deps-build.yml (hunk 4 above)
Comment (JasonMarechal25) [reply .github/workflows/centos7-system-deps-build.yml]: True

File: .github/workflows/build_centos7.yml (hunk 2 above)
Comment (JasonMarechal25) [reply .github/workflows/build_centos7.yml:128]: ¯\_(ツ)_/¯

**Pull request merged**